alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/alerts/AlertIndices.kt

```diff
@@ -151,16 +156,20 @@ class AlertIndices(
         return alertIndexInitialized && historyIndexInitialized
     }
 
-    suspend fun createAlertIndex() {
+    suspend fun createOrUpdateAlertIndex() {
         if (!alertIndexInitialized) {
             alertIndexInitialized = createIndex(ALERT_INDEX)
+        } else {
+            if (!alertIndexUpdated) updateIndexMapping(ALERT_INDEX)
         }
         alertIndexInitialized
     }
 
-    suspend fun createInitialHistoryIndex() {
+    suspend fun createOrUpdateInitialHistoryIndex() {
         if (!historyIndexInitialized) {
             historyIndexInitialized = createIndex(HISTORY_INDEX_PATTERN, HISTORY_WRITE_INDEX)
+        } else {
+            updateIndexMapping(HISTORY_WRITE_INDEX, true)
```

Comment:
Shouldn't this have the same check as `createorUpdateAlertIndex`?

Reply:
For alert history index, the index will change according to current time. So even alert config index is updated, the alert history write index may still use old schema.